client/app/hosting/ovhconfig/hosting-ovhconfig.controller.js

```diff
@@ -1,170 +1,172 @@
+                return label.replace(".", "_");
+            }
+            return "null";
+        }
+
+        initWizard () {
+            const queue = [];
+            this.loading = true;
+
+            this.User.getUrlOf("guides")
+                .then((guides) => {
+                    this.phpAppendicesGuide = guides.phpAppendices;
+                    this.hostingPhpGuide = guides.hostingPhpConfiguration;
+                });
+
+            queue.push(
+                this.hostingService.getModels()
+                    .then((apiStruct) => {
+                        this.apiStruct = {
+                            models: apiStruct.models
+                        };
+                    })
+            );
+
+            queue.push(
+                this.hostingOvhConfigService.getHistoric(this.$stateParams.productId)
+                    .then((configs) => {
+                        this.oldConfigs = configs;
+                    })
+            );
+
+            queue.push(
+                this.hostingOvhConfigService.getCurrent(this.$stateParams.productId)
+                    .then((conf) => {
+                        this.currentConfig = conf;
+                    })
+            );
+
+            return this.$q
+                .all(queue)
+                .catch(() => {
+                    this.toggle.areInitializationErrors = true;
                 })
-        );
-
-        $q.all(queue).finally(() => {
-            $scope.loading.init = false;
-            if ($scope.ovhHistoricalConfigs && $scope.ovhHistoricalConfigs.length > 0) {
-                $scope.toggle.needFirstChoice = true;
-                $scope.toggle.configFlavour = $scope.ovhHistoricalConfigs[0];
-                $scope.flavourChanged();
+                .finally(() => {
+                    this.loading = false;
+                    if (_.isArray(this.oldConfigs) && !_.isEmpty(this.oldConfigs)) {
+                        this.toggle.areOldConfigsExist = true;
+                    } else {
+                        this.toggle.areOldConfigsExist = false;
+                        this.toggle.isRollbackProcess = false;
+                        this.toggle.process = "update";
+                    }
+                });
+        }
+
+        setProcess () {
+            if (this.toggle.process === "rollback") {
+                this.toggle.isRollbackProcess = true;
+                this.selectedConfig = this.oldConfigs[0];
             } else {
-                $scope.setUpdateProcess();
+                this.selectedConfig = this.currentConfig;
             }
-        });
-    };
+            this.changeToConfig(this.selectedConfig);
+        }
 
-    $scope.checkCohesion = function () {
-        if ($scope.toggle.needFirstChoice) {
-            return;
+        changeToConfig (ovhConfig) {
+            this.clearDisplayedError();
+            this.toggle.isConfigIsEdited = false;
+            _.set(this.model, "engineName", ovhConfig.engineName);
```

Comment:
_.assign(this.model, _.pick(ovhConfig, ["engineName", "engineVersion", "environment", "httpFirewall"]));